### DISCUSSION TITLE
Error during compiling 4.4.1

### DISCUSSION THREAD

**Discussion by HudsonSchumaker** [GRRLIB/GRRLIB#14] [Q&A] [answered]
Hello, 
there was some updates in libogc (25/05/21), so i started a clean install.
when trying to install GRRLIB 4.4.1 i got this message:

`$ make clean all install
rm -f pngu.o pngu.d libpngu.a
rm -f GRRLIB_3D.o GRRLIB_bmf.o GRRLIB_bmfx.o GRRLIB_core.o GRRLIB_fbAdvanced.o GRRLIB_fileIO.o GRRLIB_gecko.o GRRLIB_print.o GRRLIB_render.o GRRLIB_snapshot.o GRRLIB_texEdit.o GRRLIB_ttf.o GRRLIB_3D.d GRRLIB_bmf.d GRRLIB_bmfx.d GRRLIB_core.d GRRLIB_fbAdvanced.d GRRLIB_fileIO.d GRRLIB_gecko.d GRRLIB_print.d GRRLIB_render.d GRRLIB_snapshot.d GRRLIB_texEdit.d GRRLIB_ttf.d libgrrlib.a

pngu.c
make[2]: powerpc-eabi-gcc: No such file or directory
make[2]: *** [pngu.o] Error 1
make[1]: *** [all] Error 1
make: *** [all] Error 1`

looking in the directory tree of devkitppc there is a directory called powerpc-eabi. I dont know if this was a change due the updates. Any help is very welcome.
Thanks in advance.

**Comment by Crayon2000** [comment, marked as answer]
I'm having some problems with devkitPPC release 39, but your error message seams more like a devkitPPC error and it's not related to GRRLIB. Can you try building another project that was working prior to the update. Maybe re-intalling devkitPPC may fix the problem. Make sure you follow the instructions on https://devkitpro.org/wiki/Getting_Started

`powerpc-eabi-gcc` should be in the `/opt/devkitpro/devkitPPC/bin` folder.

**Reply by HudsonSchumaker** [reply]
I took a look in make files of older version of GRRLIB. I putted  back 
PREFIX	:= $(DEVKITPPC)/bin/powerpc-eabi-
CC		:=	$(PREFIX)gcc
AR		:=	$(PREFIX)ar

in the make files and worked. I'm  using mac os

**Reply by Crayon2000** [reply]
This should not be required. It's done in devkitPPC's rules: https://github.com/devkitPro/devkitppc-rules/blob/master/base_tools

Make sure these environment variables are defined and it should work:

- DEVKITPRO=/opt/devkitpro
- DEVKITPPC=${DEVKITPRO}/devkitPPC

**Reply by HudsonSchumaker** [reply]
I guess the OS is not allowing the base_tools  set the variables due is being using  "export". I am using macOS Big Sur

**Reply by Crayon2000** [reply]
Did you follow all these steps to install on macOS:

1. To get the Xcode command line tools run `xcode-select --install` from Terminal.
2. Use the [pkg installer](https://github.com/devkitPro/pacman/releases/latest) to install [devkitPro pacman](https://devkitpro.org/wiki/devkitPro_pacman).
3. **Reboot your mac to get environment set.**
4. Use dkp-pacman to install your chosen tools.

Step 3 might be important 😉

**Reply by HudsonSchumaker** [reply]
Yes, I did that.

**Reply by Crayon2000** [reply]
What do you get when you type `echo $DEVKITPPC` in the console where you compile?

**Reply by HudsonSchumaker** [reply]
hudson$ echo $DEVKITPPC
/opt/devkitpro/devkitPPC
(base) sysops-MacBook-Pro-3:~ hudson$

**Reply by HudsonSchumaker** [reply]
hudson$ export
declare -x DEVKITARM="/opt/devkitpro/devkitARM"
declare -x DEVKITPPC="/opt/devkitpro/devkitPPC"
declare -x DEVKITPRO="/opt/devkitpro"
...

**Reply by Crayon2000** [reply]
If you go to `/opt/devkitpro/examples/wii/template` and type `make`, will it work?

**Reply by HudsonSchumaker** [reply]
hudson$ make
mkdir: build: Permission denied
make[1]: *** [build] Error 1
make: *** [examples] Error 1

**Reply by HudsonSchumaker** [reply]
If i use sudo, the environment entries are no considerate 

sudo make
Password:
Makefile:7: *** "Please set DEVKITPPC in your environment. export DEVKITPPC=<path to>devkitPPC".  Stop.
make: *** [examples] Error 1

**Reply by Crayon2000** [reply]
If you can't compile the template example, this issue is certainly not related to GRRLIB 😊

**Reply by HudsonSchumaker** [reply]
Yeah, you are right.
